### ISSUE TITLE
Different p-values from spatial nulls and general recommendations

### ISSUE THREAD

**Issue by rtang2100** [netneurolab/neuromaps#95]
### Description of issue

Hi Neuromaps Team,

I've been getting somewhat different p values when using different target maps for spatial nulls, which isn't too surprising. However, I do have a question about whether there's a recommendation for what map should be the target map in analyses.

I have a self-generated standardized coefficient map (L and R hemi), which I imported as an array and use construct_shape_gii to turn it into a shape gii for each hemi and used them for analyses. I have generated spatial nulls from my map, and also tried generating spatial nulls from the template maps. For example, when using my map, I got r = 0.206, p = 0.056  for correlating with cumi101 - 5-HT1a map. When using the cumi 101 for spatial nulls, I got r= 0.206, p =0.034. 

Generally speaking, I don't imagine the spatial nulls function (nulls.alexander_bloch) to act differently with shape.gii vs. other giis.  But maybe I have been wrong? One thing to note is that I have kept in mind of the ording - so when using my map's null, i have stats compare maps (my map, 5ht1a map); whereas for 5ht1a's null, i have stats stats compare maps (5ht1a map, my map).

My other question is which map should one generally use as the target map to generate spatial nulls?

Thanks!
Catherine

### Code of Conduct

- [X] I agree to follow the `neuromaps` Code of Conduct

**Comment by justinehansen** [comment]
Hi Catherine,

Re which map should be the one that gets spun: there is no single recommendation and you're right that it can make a difference although the difference in p-value is generally small. I personally try to minimize spinning maps (just because it takes time) so I spin the map that is getting repeated in many correlations. The more conservative option would be to always spin the smoother map. It's a good question though - I'm curious to hear what @VinceBaz says; Vince, do you have a different recommendation? 👀 

Re giftis: neuromaps will treat giftis the same so it's fine if your file is called "file.gii" or "file.shape.gii". The "shape" part is just indicating that the data inside the gifti is vertex-level data. Other nomenclature includes ".surf.gii" which indicates that the gifti contains the vertices and triangles of the surface mesh, or ".label.gii" which indicates that the gifti contains vertex-level labels (like for a parcellation). This just makes it easier to know the contents of the gifti but doesn't make a difference to the neuroamps function.

Hope that helps! 🙂 

Justine

**Comment by VinceBaz** [comment]
I agree with Justine on this! I tend to spin the map that I use across multiple experiments.
